crates/router/src/compatibility/stripe/payment_intents/types.rs

```diff
@@ -491,15 +509,69 @@ impl From<payments::PaymentListResponse> for StripePaymentIntentListResponse {
     }
 }
 
-#[derive(PartialEq, Eq, Deserialize, Clone)]
+#[derive(PartialEq, Eq, Deserialize, Clone, Debug)]
 #[serde(rename_all = "snake_case")]
 pub enum StripePaymentMethodOptions {
     Card {
         request_three_d_secure: Option<Request3DS>,
+        mandate_options: Option<MandateOption>,
     },
 }
 
-#[derive(Default, Eq, PartialEq, Serialize, Deserialize, Clone)]
+#[derive(PartialEq, Eq, Deserialize, Clone, Default, Debug)]
+pub struct MandateOption {
+    #[serde(default, with = "common_utils::custom_serde::iso8601::option")]
+    pub accepted_at: Option<time::PrimitiveDateTime>,
+    pub user_agent: Option<String>,
+    pub ip_address: Option<pii::Secret<String, common_utils::pii::IpAddress>>,
+    pub mandate_type: Option<String>,
```

Comment:
Can `mandate_type` be an enum

Reply:
its a string, that we will be passing from euler, so enum is not required

Reply:
done